invenio_records_permissions/policies/base.py

```diff
@@ -118,4 +119,13 @@ def query_filters(self):
         user should be able to retrieve via search.
         """
         filters = [generator.query_filter(**self.over) for generator in self.generators]
+
+        # If identity has superuser access, add match_all to filters
+        superuser_permission = self._expand_action(superuser_access)
+        superuser_needs = superuser_permission.needs if superuser_permission else set()
+        identity = self.over.get("identity")
+        identity_provides = identity.provides if identity else set()
+        if superuser_needs & identity_provides:
+            filters.append(Q("match_all"))
+
```

Comment:
This assumes that the way to set a superuser is via the setup of ActionRole or ActionUser... This is how needs and excludes get it, so the same mechanism is used, but it's surprising that it's not just a check for the `superuser_access` Need directly. I remember some vague performance considerations, but I don't understand how checking for this one Need isn't performant enough... :thinking:  

To be candid, permissions are hard and it would be worth revisiting some of our mechanics.